### PR TITLE
Android wrapper: Modernize build files

### PR DESCRIPTION
**Pull request by okarmazin**
This PR updates the Android build files and updates project dependencies. No source files were changed.

I tried to introduce this change by incremental atomic commits to keep the build file history tracked by Git, but was unable to do so. The amount of changes required for the Kotlin script conversion was too great.

Notable changes:
- Convert build scripts to Kotlin (build.gradle -> build.gradle.kts)
- Kotlin 1.9.10
- Gradle 8.3
- Use Gradle Version Catalog
- Android Gradle Plugin 8.1.1

### CODE REVIEW

**Comment by okarmazin** [comment]
Supersedes #555

**Comment by axxel** [comment]
The main road-block that prevented #555 from moving forward is still there: https://github.com/zxing-cpp/zxing-cpp/actions/runs/6262637658/job/17005466542?pr=619#step:3:34

**Comment by okarmazin** [comment]
Updated the Java version used by the CI worker, should build now.

**Comment by axxel** [comment]
Thanks, but apparently there is still something missing: https://github.com/zxing-cpp/zxing-cpp/actions/runs/6262900310/job/17006568650?pr=619

**Comment by okarmazin** [comment]
Right... Didn't know it was a required field.

File: wrappers/android/gradle/libs.versions.toml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+[versions]
```
Comment (axxel) [inline wrappers/android/gradle/libs.versions.toml:1]: Is this toml style variable collection the accepted best practice nowadays or your personal preference?

File: wrappers/android/gradle/wrapper/gradle-wrapper.jar
Comment (axxel) [inline wrappers/android/gradle/wrapper/gradle-wrapper.jar:1]: Adding a binary jar file seems odd. What is its purpose?

**Review by axxel — COMMENTED** [review]
Thanks for your contribution. Much appreciated.

File: wrappers/android/gradle/wrapper/gradle-wrapper.jar
Comment (okarmazin) [reply wrappers/android/gradle/wrapper/gradle-wrapper.jar:1]: `gradle-wrapper.jar` together with the scripts `gradlew` and `gradlew.bat` are the output of Gradle itself. I did not change these files, Gradle did so itself when its version was upgraded to 8.3.

These changes result from running `./gradlew wrapper --gradle-version=8.3`.

[Gradle Wrapper documentation](https://docs.gradle.org/current/userguide/gradle_wrapper.html)

Comment (benjohnde) [reply wrappers/android/gradle/wrapper/gradle-wrapper.jar:1]: The wrapper is normally included in the gradle directory and considered as best practise, although I do agree, seeing binary files in a repo raises an odd feeling :) The wrapper is used to run gradle and - if necessary - to download the specified gradle version to have some sort of reproducible build environment (at least from the build tool perspective).

File: wrappers/android/app/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,50 @@
+plugins {
```
[the remark this answers is not in the document]
Comment (benjohnde) [reply wrappers/android/app/build.gradle.kts:1]: Nowadays Kotlin is preferred over Groovy to write these build scripts for gradle. `build.gradle` implies Groovy whereas `build.gradle.kts` implies the use of Gradle's Kotlin DSL, cf. https://docs.gradle.org/current/userguide/migrating_from_groovy_to_kotlin_dsl.html.

**Comment by benjohnde** [comment]
Thanks for the contribution @okarmazin. I still want to overhaul the iOS wrapper someday very soon :) That's a nice push!

File: wrappers/android/app/build.gradle.kts (hunk 2 above)
[the remark this answers is not in the document]
Comment (okarmazin) [reply wrappers/android/app/build.gradle.kts:1]: I can do it no better justice than Gradle itself:

https://blog.gradle.org/kotlin-dsl-is-now-the-default-for-new-gradle-builds

https://developer.android.com/build/migrate-to-kotlin-dsl

A note about the sentence "...builds using Kotlin tend to be slower than builds using Groovy..." from the Android link. This is _not_ a factor in this project! It refers to very large projects with tens of modules and thousands of lines of complex build scripts.

[the remark this answers is not in the document]
Comment (okarmazin) [reply wrappers/android/app/build.gradle.kts:1]: It's difficult to convey how much superior the editing experience is with the Kotlin DSL.

[the remark this answers is not in the document]
Comment (axxel) [reply wrappers/android/app/build.gradle.kts:1]: Awesome. Thanks to both of you for the very convincing motivation :).

**Pull request merged**